plugins/vst_base/VstPlugin.cpp

```diff
@@ -91,20 +147,32 @@ VstPlugin::VstPlugin( const QString & _plugin ) :
 	m_embedMethod( gui
 			? ConfigManager::inst()->vstEmbedMethod()
 			: "headless" ),
-	m_badDllFormat( false ),
 	m_version( 0 ),
 	m_currentProgram()
 {
 	setSplittedChannels( true );
 
-	tryLoad( REMOTE_VST_PLUGIN_FILEPATH );
-#ifdef LMMS_BUILD_WIN64
-	if( m_badDllFormat )
+	PE::MachineType machineType;
+	try {
+		PE::FileInfo peInfo(_plugin);
+		machineType = peInfo.machineType();
+	} catch (std::runtime_error& e) {
+		qCritical() << "Error while determining PE file's machine type: " << e.what();
+		machineType = PE::MachineType::unknown;
+	}
+
+	switch(machineType)
 	{
-		m_badDllFormat = false;
-		tryLoad( "32/RemoteVstPlugin32" );
+	case PE::MachineType::amd64:
+		tryLoad( "RemoteVstPlugin64" );
```

Comment:
If I understand correctly, this will try to load 64bit DLL in 32bit system using nonexistent `RemoteVstPlugin64`. It will print an error message, but Windows users can't see that.
Do we need to try this on 32bit LMMS?

Reply:
This is correct, there's no visible error message in the GUI when loading fails. This PR doesn't change that.

Reply:
I think error message issue can be fixed later, but do we ever need to try to load VST with `RemoteVstPlugin64` on 32bit systems?

Reply:
Theoretically, we could ship a `RemoteVstPlugin64` with our 32bit Windows installers for users who install 32bit LMMS because they're unaware of the difference. In that case, it's good to try `RemoteVstPlugin64`. In any other case, there's no harm done here. `tryLoad` will discover that `RemoteVstPlugin64` doesn't exist and will abort, at which point we can easily add an error message later.

Reply:
Got it, I didn't think about that.

Reply:
I think we may try to detect processor in runtime if possible, though it looks like out of scope.
Qt provides [`QSysInfo::currentCpuArchitecture`](http://doc.qt.io/qt-5/qsysinfo.html#currentCpuArchitecture) as of Qt 5.4, and it's also possible to use OS-dependent detection.

Reply:
I opened #4447 to track it.